app/views/shell/_top.html.erb

```diff
@@ -51,7 +51,16 @@
   <% unless internal_navigation? %>
     <% cache(release_adjusted_cache_key("top-html-and-config--#{user_signed_in?}")) do %>
       <body data-user-status="<%= user_logged_in_status %>" class="<%= SiteConfig.default_font.tr("_", "-") %>-article-body" data-pusher-key="<%= ApplicationConfig["PUSHER_KEY"] %>">
-        <div id="body-styles"></div>
+        <div id="body-styles">
+          <style>
+            :root {
+              --accent-brand: <%= SiteConfig.primary_brand_color_hex %>;
+              --accent-brand-darker: <%= HexComparer.new([SiteConfig.primary_brand_color_hex]).brightness(0.85) %>;
+              --accent-brand-lighter: <%= HexComparer.new([SiteConfig.primary_brand_color_hex]).brightness(1.1) %>;
+              --accent-brand-a10: #{rgba(<%= SiteConfig.primary_brand_color_hex %>, 0.1)};
+            }
+          </style>
```

Comment:
I'm not sure if that's correct but it seems like THIS will be replaced in [certain cases](https://github.com/forem/forem/blob/d86b9d9ce5b62535c613343b57262262fc1d125d/app/views/layouts/_user_config.html.erb#L9-L15).  Is that intended/expected?

Reply:
Another thing: are we sure that updating the brightness is ENOUGH to make lighter/darker variants look good? Because this will look different when your brand color is #000 vs some shade of purple vs some shade of light blue.

Reply:
Yeah, my thoughts are that the default theme can be branded but the other themes don't need the brand colors because they may clash.

That makes sense about the brightness needing to vary depending on the hex.... We could likely programmatically determine the darkness of a hex and adjust our adjustment based on the root darkness. Given that we're only accepting one admin-generated field here and then working from it I'd say it's okay if we ship it without having all the accents worked out and then seek to get that right after the fact?

Reply:
makes sense 👍